### ISSUE TITLE
NewsletterProfile.Get whitescreen under PHP8

### ISSUE THREAD

**Issue by pbatroff** [systopia/de.systopia.newsletter#20]
With PHP8 the following error message is thrown when using NewsletterProfile.Get:

```
Got error 'PHP message: TypeError: array_filter(): Argument #1 ($array) must be of type array, null given in /home/phil/Documents/src/de.systopia.newsletter/api/v3/NewsletterProfile/Getsingle.php on line 51 #0 /home/phil/Documents/src/de.systopia.newsletter/api/v3/NewsletterProfile/Getsingle.php(51): array_filter()\n#1 /srv/http/latest9/drupal/vendor/civicrm/civicrm-core/Civi/API/Provider/MagicFunctionProvider.php(89): civicrm_api3_newsletter_profile_getsingle()\n#2 /srv/http/latest9/drupal/vendor/civicrm/civicrm-core/Civi/API/Kernel.php(149): Civi\\API\\Provider\\MagicFunctionProvider->invoke()\n#3 /srv/http/latest9/drupal/vendor/civicrm/civicrm-core/Civi/API/Kernel.php(81): Civi\\API\\Kernel->runRequest()\n#4 /srv/http/latest9/drupal/vendor/civicrm/civicrm-core/api/api.php(133): Civi\\API\\Kernel->runSafe()\n#5 /home/phil/Documents/src/de.systopia.newsletter/api/v3/NewsletterProfile/Get.php(27): civicrm_api3()\n#6 /srv/http/latest9/drupal/vendor/civicrm/civicrm-core/Civi/API/Provider/MagicFunctionProvider.php(89): civi...', referer: http://localhost/latest9/public/civicrm/api3
```

This doesn't seem to happen under php7.4, and the origin of the code is from [this](https://github.com/systopia/de.systopia.newsletter/commit/23d4177f85f12946e0242b37f9cd3af6e198c188) commit.

Often, but not necessarily always in this place https://github.com/systopia/de.systopia.newsletter/blob/103214b897edc09e3d65e1eaeb1409677d4ac3d9/api/v3/NewsletterProfile/Getsingle.php#L52 
$field['options'] isn't set yet, thus throwing the php error.

**Comment by pbatroff** [comment]
Tested this and works as intended. not in production yet though

**Comment by jensschuppe** [comment]
This has been released with version `1.0-beta1`.
